File: Flutter-Harpia-Health-Analysis/lib/model/firebase/FcmMessage.dart
import 'package:flutter_harpia_health_analysis/model/firebase/FcmNotification.dart';

import 'FcmData.dart';

class FcmMessage {
  // late FcmNotification _notification;
  late FcmData _fcmData;
  FcmMessage({required fcmData}) {
    _fcmData = fcmData;
  }

  /*FcmMessage({required notification, required fcmData}) {
    _notification = notification;
    _fcmData = fcmData;
  }*/

  factory FcmMessage.fromJson(Map<String, dynamic> json) {
    return FcmMessage(
        // notification: FcmNotification.fromJson(json),
        fcmData: FcmData.fromJson(json));
  }


  @override
  String toString() {
    return 'FcmMessage{_fcmData: $_fcmData}';
  }

  FcmData get fcmData => _fcmData;

  // FcmNotification get notification => _notification;
}

File: ws/src/main/java/com/ahmeteminsaglik/ws/model/firebase/FcmMessage.java
package com.ahmeteminsaglik.ws.model.firebase;

public class FcmMessage {

    private String to;
    private FcmNotification notification;
    private FcmData data;

    public String getTo() {
        return to;
    }

    public void setTo(String to) {
        this.to = to;
    }

    public FcmNotification getNotification() {
        return notification;
    }

    public void setNotification(FcmNotification notification) {
        this.notification = notification;
    }

    public FcmData getData() {
        return data;
    }

    public void setData(FcmData data) {
        this.data = data;
    }

    @Override
    public String toString() {
        return "FcmMessage{" +
                "to='" + to + '\'' +
                ", notification=" + notification +
                ", data=" + data +
                '}';
    }
}

